pkg/network/ebpf/c/tracer-events.h

```diff
@@ -14,7 +14,29 @@ static __always_inline int get_proto(conn_tuple_t *t) {
     return (t->metadata & CONN_TYPE_TCP) ? CONN_TYPE_TCP : CONN_TYPE_UDP;
 }
 
+static __always_inline void clean_protocol_classification(conn_tuple_t *tup) {
+    conn_tuple_t conn_tuple = *tup;
+    conn_tuple.pid = 0;
+    conn_tuple.netns = 0;
+    bpf_map_delete_elem(&connection_protocol, &conn_tuple);
+
+    conn_tuple_t *skb_tup_ptr = bpf_map_lookup_elem(&conn_tuple_to_socket_skb_conn_tuple, &conn_tuple);
+    if (skb_tup_ptr != NULL) {
```

Comment:
Do you think some telemetry is needed here as well, since this should never be NULL?

Reply:
so it can happen on some cases
open connection, close it

the socket filter won't do anything since it is empty payloads only